### PR TITLE
4605 bug fix double log button

### PR DESCRIPTION
**Pull request by emitche**
- Remove add log button at bottom of logs list
- Ensure last deleted log returns view to no logs view

### CODE REVIEW

File: app/assets/javascripts/backbone/views/project_step_view.coffee [hunk 1]
```diff
@@ -91,5 +91,15 @@ class MS.Views.ProjectStepView extends Backbone.View
 
   deleteLog: (e, response) ->
     $.post @$(e.target).attr('href'), {_method: 'DELETE'}
+
+    # Number of logs currently visible includes one just deleted from database
+    logsList = @$(e.target).closest('.logs-list')
+    numOfLogs = @$(logsList).children().length
```
Comment (hooverlunch) [inline app/assets/javascripts/backbone/views/project_step_view.coffee]: Please combine the two above lines into one. It's ok to chain a bunch of jquery methods in a row (it's designed for this), as long as readability doesn't suffer.

`@$(e.target).closest('.logs-list').children().length` is quite clear.

File: app/assets/javascripts/backbone/views/project_step_view.coffee [hunk 2]
```diff
@@ -91,5 +91,15 @@ class MS.Views.ProjectStepView extends Backbone.View
...
+
+    if numOfLogs < 2
+      logContainer = @$(logsList).closest('.step-logs')
+      @$(logContainer).removeClass('expanded')
+      @$(logContainer).find('.view-logs-ctrl').html('<span class="no-logs">No Logs</span>')
```
Comment (hooverlunch) [inline app/assets/javascripts/backbone/views/project_step_view.coffee]: Better to show/hide the existing than to insert one like this. The HTML you're inserting is not translated, for one. This is why it's best to keep HTML out of JS if at all possible.

**Comment by emitche** [comment]
Best to rerender entire logs partial instead.

**Comment by emitche** [comment]
The way this is currently set up, if the additional add log button was conditional on number of logs present, it would be easy to implement through a variable in the controller.

**Comment by hooverlunch** [comment]
GREAT WORK. Much better!

I made one commit simplifying the coffeescript. Notice that you don't need to list out all JS method params if you aren't using them. You can cut off the unused ones on the end.

**Comment by emitche** [comment]
Smart. Thanks!

**Comment by hooverlunch** [comment]
Going to merge this now as I don't want conflict in other issue I'm working on :)

**Pull request merged**